### ISSUE TITLE
Debug unlocking all recipes does not unlock all recipes

### ISSUE THREAD

**Issue by NetSysFire** [CleverRaven/Cataclysm-DDA#77857]
### Describe the bug

Title. This is a regression.

### Attach save file

n/a

### Steps to reproduce

1. You want to look up a book-only recipe for debugging, in this case chainmail.
2. Go into the debug menu > player > unlock all recipes
3. The recipes are, in fact, not unlocked, despite all skills also being at 10.

### Expected behavior

The recipes get unlocked.

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: cdda-experimental-2024-11-14-1240 65b3721 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

**Comment by ZhilkinSerg** [comment]
Is it a nested recipe? They are never learned I think.

**Comment by NetSysFire** [comment]
Reading the book once for 5 minutes does unlock them permanently though.

**Comment by kevingranade** [comment]
Issues about debug features are not valid.

**Comment by NetSysFire** [comment]
So where do we track them?

**Comment by ZhilkinSerg** [comment]
The idea is you just fix whatever you found.

**Comment by NetSysFire** [comment]
If I would be able to fix that, I had. I don't do C++. And simply ignoring issues like that is making matters worse, Kevin. I have seen you do it a bunch in the recent times, and I heavily disapprove of that.

**Comment by ZhilkinSerg** [comment]
The idea is something not working in debug menu does not affect too many people as it is a developer tool not a gameplay feature.

**Comment by NetSysFire** [comment]
I am well aware of that, but it is an *essential* tool to contribute content, hunt bugs and the like, as you know. Ignoring this entirely and automatically making any bug about it invalid is a very poor choice of action.

**Comment by NetSysFire** [comment]
This is actively hindering people to audit content as there is no better place to view e.g rendered recipes. Requring to spawn every book and read them one by one is extremely tedious.

Given the fact that other issues about the debugging menu were made in the meantime and none were closed plus no further clarification on the reasoning was given, I am reopening this.

**Comment by kevingranade** [comment]
We have thousands of open issues which is leading to a *severe* problem with tracking more critical issues,  the answer is triaging away lower priority issues to allow better focus on the more critical ones.

**Comment by NetSysFire** [comment]
True "Low priority" bugs worth closing are typos, minor mapgen issues and similar easy to fix things that barely impact gameplay and contributing. And "triaging them away" is applying a label, not closing them. And calling confirmed bugs "invalid", which this one most certainly is not, does not make them go away. Closing valid bug reports like these leads to them being almost invisible, encouraging duplicates and actually eating *more* triage time.

Focusing on the more critical ones happens through applying priority labels, milestones (we do not have one for that yet but it sounds sensible to possibly create one) + project boards and pinning issues. I am behind pinning the majority of recently pinned issues as other triagers, like you, have not made use of them enough to direct priority.

Why do I have to argue with you about a clearly valid bug? All this time spent playing these pointless social games of yours is taking time away from actually triaging. Stop closing this one and if you really, really feel about this so strongly, I advise you to apply consistency instead of, as I currently perceive it, only hyperfocusing on the issues I created and ignoring all others that *actually* need closing according to your vague definition of "low priority bug" that is not written down anywhere.

Also, given how heavily we rely on contributions and how many recipes need auditing, I can very well see this being medium. This is not about changing some value. This is not a trivial json fix. This is impacting contributing. If I could fix them, I would not have bothered writing an issue.

**Comment by Procyonae** [comment]
I can't find why but for some reason only some of them are being immediately unlocked while the rest unlock on save->load which is at least a work around. I'm don't think this bug is exclusive to debug either but I'm not sure how to easily test that.

**Comment by IdleSol** [comment]
I have a question. After you unlock recipes, do you skip a move or do you look right away?

**Comment by NetSysFire** [comment]
This is most curious indeed and I tried a few times and this is very inconsistent. In this case I am after the chainmail recipes as I have someone here that actually works on chainmail for a living.

New debug character, unlock all recipes? On the same turn, nothing. Waiting a turn? Also nothing. This randomly happens in the next 10-30 turns, the timespan seems to vary a lot.

**Comment by IdleSol** [comment]
What exactly is the recipe? I'd like to try it at my place.

**Comment by NetSysFire** [comment]
Just general chainmail recipes, the ones you assemble from scratch using wires and links, which take a looot of time. Right now only the "assemble multiple pieces of already finished chainmail" are shown by default as they do not need a book to figure out.

**Comment by IdleSol** [comment]
There are too many of these recipes. I've never used chain mail. So I don't know exactly how many recipes there should be. So I'm asking for one recipe that should definitely be there, but that doesn't show up.

I checked with `tempered steel chainmail l`eggings. When increasing skills to 10 the recipe is not there. After unlocking and skipping a move, the recipe is there. In 3 out of 3.

![1](https://github.com/user-attachments/assets/ac9adbd2-6ef6-4bbb-bed1-36fa42757b38)

Screenshot from 4 tries: I didn't upgrade the skill to 10, instead I unlocked the recipes immediately.
